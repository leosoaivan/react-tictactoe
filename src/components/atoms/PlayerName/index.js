import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Root = styled.div`
  box-sizing: border-box;
  font-size: 3em;
  font-family: 'Montserrat', sans serif;
  font-weight: 600;
  margin-left: 0.5em;
  margin-right: 0.5em;
  min-width: 0;
  max-width: calc(100% - 0.5em);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: no-wrap;
`;

const PlayerName = ({ className, name, symbol }) => {
  const displayedName = (name ? `${name} (${symbol})` : `(${symbol})`);

  return (
    <Root className={className}>
      {displayedName}
    </Root>
  );
};

PlayerName.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  symbol: PropTypes.string,
};

PlayerName.defaultProps = {
  className: null,
  name: '',
  symbol: '',
};

export default PlayerName;
