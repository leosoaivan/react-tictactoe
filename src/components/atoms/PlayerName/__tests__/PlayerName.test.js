import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { createSerializer } from 'enzyme-to-json';
import PlayerName from '..';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('PlayerName', () => {
  it('Renders a PlayerName component with a name', () => {
    const tree = renderer.create(
      <PlayerName name="Player One" symbol="X" />,
    ).toJSON();
    const displayedName = tree.children[0];

    expect(tree).toMatchSnapshot();
    expect(displayedName).toEqual('Player One (X)');
  });

  it('Renders a PlayerName component without a name', () => {
    const tree = renderer.create(
      <PlayerName symbol="X" />,
    ).toJSON();
    const displayedName = tree.children[0];

    expect(tree).toMatchSnapshot();
    expect(displayedName).toEqual('(X)');
  });
});
