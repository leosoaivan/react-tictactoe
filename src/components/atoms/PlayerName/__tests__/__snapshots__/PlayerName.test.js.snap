// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerName Renders a PlayerName component with a name 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 3em;
  font-family: 'Montserrat',sans serif;
  font-weight: 600;
  margin-left: 0.5em;
  margin-right: 0.5em;
  min-width: 0;
  max-width: calc(100% - 0.5em);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: no-wrap;
}

<div
  className="c0"
>
  Player One (X)
</div>
`;

exports[`PlayerName Renders a PlayerName component without a name 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 3em;
  font-family: 'Montserrat',sans serif;
  font-weight: 600;
  margin-left: 0.5em;
  margin-right: 0.5em;
  min-width: 0;
  max-width: calc(100% - 0.5em);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: no-wrap;
}

<div
  className="c0"
>
  (X)
</div>
`;
